doc/api/assert.markdown

```diff
@@ -85,9 +85,27 @@ Custom error validation:
       "unexpected error"
     );
 
-## assert.doesNotThrow(block[, message])
+## assert.doesNotThrow(block[, error][, message])
 
-Expects `block` not to throw an error, see `assert.throws` for details.
+Expects `block` not to throw an error. See `assert.throws` for more details.
+
+If `block` throws an error and if it is of a different type from `error`, the thrown error will get propagated back to the caller. The following call will throw the `TypeError`, since we're not matching the error types in the assertion.
+
+    a.doesNotThrow(
+      function() {
+        throw new TypeError("Wrong value");
+      },
+      a.AssertionError
+    );
```

Comment:
For consistency with the the other sample code in the doc, it might be best to use `assert.whatever` rather than `a.whatever`.